adapters/celestia/src/verifier/proofs.rs

```diff
@@ -45,24 +45,27 @@ impl CompletenessProof {
 pub struct CorrectnessProof(pub Vec<EtxProof>);
 
 impl CorrectnessProof {
-    pub fn for_block(block: &FilteredCelestiaBlock, relevant_txs: &Vec<BlobWithSender>) -> Self {
+    pub fn for_block(
+        block: &FilteredCelestiaBlock,
+        blobs: &[<CelestiaSpec as sov_rollup_interface::da::DaSpec>::BlobTransaction],
+    ) -> Self {
         let mut needed_tx_shares = Vec::new();
 
         // Extract (and clone) the position of each transaction
-        for tx in relevant_txs.iter() {
-            let commitment = recreate_commitment(block.square_size(), BlobRef::with(&tx.blob.0))
-                .expect("commitment is valid");
-
-            let (_, position) = block
-                .relevant_pfbs
-                .get(&commitment[..])
-                .expect("commitment must exist in map");
-            needed_tx_shares.push(position.clone());
+        for tx in blobs.iter() {
+            // We process the transaction only if we read something from it
```

Comment:
On closer review there might be one more soundness issue here. In our system, an STF might decide to ignore a transaction based on its sender - so even if the data is unread, we still need to check that the PFB transaction was provided correctly - we just don't need to open the blob commitment

Reply:
After discussion with Preston I have opened the issue #412 to implement the partial blob optimization for Celestia.